restcomm/restcomm.sms/src/main/java/org/restcomm/connect/sms/smpp/SmppMessageHandler.java

```diff
@@ -157,7 +157,10 @@ public void onReceive(Object message) throws Exception {
             if (sms == null) {
                 logger.warning("responseMessageId=" + smppMessageId + " was never received!");
             } else {
-                this.storage.getSmsMessagesDao().updateSmsMessage(sms.setSmppMessageId(null).setStatus(deliveryStatus));
+                sms.setSmppMessageId(null);
```

Comment:
CAREFUL!! each setter returns a brand new SMS object

Reply:
ummm,right this is a bug for sure. I would say lets create a builder with copy constructor instead, this is very dangerous idiom we are using.